python/dataverse_sdk/schemas/client.py

```diff
@@ -68,10 +69,65 @@ class Config:
         use_enum_values = True
 
 
+class DataSource(str, Enum):
+
+    Azure = "azure"
+    AWS = "aws"
+    LOCAL = "local"
+
+
+class DataConfig(BaseModel):
+    storage_url: str
+    container_name: Optional[str]
+    sas_token: Optional[str]
+    data_folder: str
+    sequential: bool
+    generate_metadata: bool
+    description: Optional[str]
+    type: str
+    annotation_format: str
+
+    class Config:
+        extra = "allow"
+
+
 class Project(BaseModel):
     id: Optional[int] = None
     name: str
     description: Optional[str] = None
     ego_car: Optional[str] = None
     ontology: Ontology
     sensors: list[Sensor]
+    client: Optional[object] = None
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def create_dataset(self, name: str, source: DataSource, dataset: DataConfig):
+        if self.client is not None:
+            dataset_output = self.client.create_dataset(
+                name=name, source=source, project=self, dataset=dataset
+            )
+            return dataset_output
+        else:
+            raise NotImplementedError("ClientServer is not defined")
+
+
+class Dataset(BaseModel):
+    id: Optional[int] = None
+    name: str
+    data_source: str
+    annotation_format: str
+    sequential: Optional[bool]
+    generate_metadata: Optional[bool]
+    description: Optional[str] = None
+    project: Optional[dict] = None
+    type: str
+    file_count: Optional[int] = None
+    image_count: Optional[int] = None
+    pcd_count: Optional[int] = None
+    created_by: Optional[int] = None
+    client: Optional[object] = None
+    container_name: Optional[str] = None
+    storage_url: Optional[str] = None
+    status: str
```

Comment:
Seems like still missing some of the classes e.g. `DatasetType`, `DatasetStatus`, `AnnotationFormat` to be implemented.

```suggestion
class Dataset(BaseModel):
    id: Optional[int] = None
    project: Project
    name: str
    type: DatasetType
    data_source: DataSource
    status: DatasetStatus
    annotation_format: AnnotationFormat
    sequential: bool = False
    generate_metadata: bool = False
    description: Optional[str] = None
    file_count: Optional[int] = None
    image_count: Optional[int] = None
    pcd_count: Optional[int] = None
    container_name: Optional[str] = None
    storage_url: Optional[str] = None
```